ArchiSteamFarm.OfficialPlugins.Monitoring/MonitoringPlugin.cs

```diff
@@ -0,0 +1,189 @@
+//   / _ \  | '__|/ __|| '_ \ | |\___ \ | __|/ _ \ / _` || '_ ` _ \ | |_  / _` || '__|| '_ ` _ \
+//  / ___ \ | |  | (__ | | | || | ___) || |_|  __/| (_| || | | | | ||  _|| (_| || |   | | | | | |
+// /_/   \_\|_|   \___||_| |_||_||____/  \__|\___| \__,_||_| |_| |_||_|   \__,_||_|   |_| |_| |_|
+// ----------------------------------------------------------------------------------------------
+// |
+// Copyright 2015-2024 Łukasz "JustArchi" Domeradzki
+// Contact: JustArchi@JustArchi.net
+// |
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// |
+// http://www.apache.org/licenses/LICENSE-2.0
+// |
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Metrics;
+using System.Linq;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using ArchiSteamFarm.Core;
+using ArchiSteamFarm.IPC.Integration;
+using ArchiSteamFarm.Plugins;
+using ArchiSteamFarm.Plugins.Interfaces;
+using ArchiSteamFarm.Steam;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Metrics;
+using SteamKit2;
+
+namespace ArchiSteamFarm.OfficialPlugins.Monitoring;
+
+[Export(typeof(IPlugin))]
+[SuppressMessage("ReSharper", "MemberCanBeFileLocal")]
+internal sealed class MonitoringPlugin : OfficialPlugin, IWebServiceProvider, IGitHubPluginUpdates, IDisposable {
+	private const string MeterName = SharedInfo.AssemblyName;
+
+	private const string MetricNamePrefix = "asf";
+
+	private static bool Enabled => ASF.GlobalConfig?.IPC ?? false;
+
+	[JsonInclude]
+	[Required]
+	public override string Name => nameof(MonitoringPlugin);
+
+	/// <inheritdoc />
+	public string RepositoryName => SharedInfo.GithubRepo;
+
+	[JsonInclude]
+	[Required]
+	public override Version Version => typeof(MonitoringPlugin).Assembly.GetName().Version ?? throw new InvalidOperationException(nameof(Version));
+
+	private Meter? Meter;
+
+	public void Dispose() => Meter?.Dispose();
+
+	public void OnConfiguringEndpoints(IApplicationBuilder app) {
+		ArgumentNullException.ThrowIfNull(app);
+
+		if (!Enabled) {
+			return;
+		}
+
+		app.UseEndpoints(static builder => builder.MapPrometheusScrapingEndpoint());
+	}
+
+	public void OnConfiguringServices(IServiceCollection services) {
+		ArgumentNullException.ThrowIfNull(services);
+
+		if (!Enabled) {
+			return;
+		}
+
+		services.AddOpenTelemetry().WithMetrics(
+			builder => {
+				InitializeMeter();
```

Comment:
Extract this outside of builder, after `Enabled` check, since we don't need `builder` for it.